misk/src/main/kotlin/misk/clustering/lease/Lease.kt

```diff
@@ -18,4 +18,24 @@ interface Lease {
    * so it is marked as a function rather than a property to make the potential expense clearer
    */
   fun checkHeld(): Boolean
+
+  /**
+   * Registers a listener that is called on lease state changes.
+   */
+  fun addListener(listener: StateChangeListener)
```

Comment:
Should there be a `removeListener()` method? I didn't see the need for it and preferred not to make the interface wider.

Reply:
Agreed, don't think it's necessary. Can always add it later if we do see the need, but I think listeners will always be registered at startup and never removed.